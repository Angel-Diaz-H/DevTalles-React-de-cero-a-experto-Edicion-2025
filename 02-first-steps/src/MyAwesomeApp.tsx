import type { CSSProperties } from "react";

const myStyle: CSSProperties = {
    backgroundColor: '#c1e4c1ff',
    borderRadius: 30,
    padding: 10,
    margin: 15,
    textAlign: 'center',
}

const firstName = 'Angel';
const lastName = 'Díaz';

const favoriteGame = ['Elden Ring', 'Smash', 'Metal Gear']
const isActive = false;

const address = {
    zipCode: 'ABC-123',
    country: 'Canadá'
};

export function MyAwesomeApp() {
    return (
        <>
            <h1>{firstName}</h1>
            <h3>{lastName}</h3>

            <p>{favoriteGame.join(', ')}</p>
            <p>2+2 = {2 + 2}</p>

            {/* No existe una representación visual de React para boolean. */}
            <h1>{isActive}</h1>
            <h1>{isActive ? 'Sí' : 'No'}</h1>

            <p style={myStyle}>
                {JSON.stringify(address)}
            </p>
        </>
    )
};