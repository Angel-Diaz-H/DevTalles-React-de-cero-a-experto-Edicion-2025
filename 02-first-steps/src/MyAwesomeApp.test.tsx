import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import { render, screen } from '@testing-library/react'

describe('MyAwesomeApp', () => {
    test('Should render firstName and lastName', () => {
        const { container } = render(<MyAwesomeApp />);
        screen.debug()

        const h1 = container.querySelector('h1');
        const h3 = container.querySelector('h3');

        expect(h1?.innerHTML).toContain('Angel');
        expect(h3?.innerHTML).toContain('DÃ­az');
    });

    test('Should render firstName and lastName - screen', () => {
        render(<MyAwesomeApp />)
        screen.debug()
        const h1 = screen.getByTestId('first-name-title');
        expect(h1.innerHTML).toContain('Angel');
    });

    test('Should match snapshot', () => {
        const { container } = render(<MyAwesomeApp />);
        expect(container).toMatchSnapshot();
    });

    test('Should match snapshot', () => {
        render(<MyAwesomeApp />);
        expect(screen.getByTestId('div-app')).toMatchSnapshot();
    });
});