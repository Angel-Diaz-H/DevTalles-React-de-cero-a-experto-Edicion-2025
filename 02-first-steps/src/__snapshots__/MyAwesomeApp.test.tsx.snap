// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > Should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Angel
    </h1>
    <h3>
      Díaz
    </h3>
    <p>
      Elden Ring, Smash, Metal Gear
    </p>
    <p>
      2+2 = 
      4
    </p>
    <h1 />
    <h1>
      No
    </h1>
    <p
      style="background-color: rgb(193, 228, 193); border-radius: 30px; padding: 10px; margin: 15px; text-align: center;"
    >
      {"zipCode":"ABC-123","country":"Canadá"}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > Should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Angel
  </h1>
  <h3>
    Díaz
  </h3>
  <p>
    Elden Ring, Smash, Metal Gear
  </p>
  <p>
    2+2 = 
    4
  </p>
  <h1 />
  <h1>
    No
  </h1>
  <p
    style="background-color: rgb(193, 228, 193); border-radius: 30px; padding: 10px; margin: 15px; text-align: center;"
  >
    {"zipCode":"ABC-123","country":"Canadá"}
  </p>
</div>
`;
